### ISSUE TITLE
Inconsistent behavior of `move left|right|up|down`

### ISSUE THREAD

**Issue by mikkeloscar** [i3/i3#2257]
Output of `i3 --moreversion 2>&- || i3 --version`:

```
Binary i3 version:  4.12 (2016-03-06, branch "4.12") © 2009 Michael Stapelberg and contributors
Running i3 version: 4.12 (2016-03-06, branch "4.12") (pid 4108)
Loaded i3 config: /home/moscar/.config/i3/config (Last modified: 2016-03-12T08:47:40 CET, 29177 seconds ago)

The i3 binary you just called: /usr/bin/i3
The i3 binary you are running: i3
```

URL to a logfile as per http://i3wm.org/docs/debugging.html:

I have not included a log because I don't know when the error will happen (as described below), thus the log will be huge and might include a lot of private info, which I would rather not include here (or try to remove from the possible relevant parts of the log).

If this is a must for the issue to be considered, then I will try to see what I can do, but it might take some time.

**What I did:**

I originally described the bug here: SirCmpwn/sway#500

When using the commands `move left|right|up|down` to move a focused window from one output to another, the expected behavior is to have the focus follow the window to the new output. This always happen when I first launch i3, but at some point i3 will be in a state where the focus doesn't follow a window to a new output when using the `move left|right|up|down` commands.

I have tried to find a way to reproduce the issue reliably but I can't figure out what I'm doing to make i3 enter this broken state. It just seem to happen after I have been using i3 for a while.

I can always solve the issue temporarily by restarting i3 in place. After a restart it will work as expected until it somehow gets to a broken state again.

I know this isn't much to go by, so I have tried to capture it in this screencapture: https://www.youtube.com/watch?v=Ri7B7hLv3d4

I started recording when i3 was in a broken state, and as you can see, the focus doesn't follow the window when I move it from output to output. At [0:14](https://youtu.be/Ri7B7hLv3d4?t=14) I restart i3 in place and after that it works as expected.

**What I saw:**

Described above

**What I expected instead:**

I expect the commands `move left|right|up|down` to always maintain focus when moving a window between outputs.

As you can see from SirCmpwn/sway#500 I discovered the issue when implementing the `move left|right|up|down` commands in [sway](https://github.com/SirCmpwn/sway) and we can't merge that before we know that it is in fact a bug in i3. So I'd be grateful if you could just confirm that this is a bug, even though it might be a challenge to debug and fix.

**Comment by i3bot** [comment]
I don’t see a link to logs.i3wm.org. Did you follow http://i3wm.org/docs/debugging.html? (In case you actually provided a link to a logfile, please ignore me.)

**Comment by stapelberg** [comment]
Thanks for your patience with this issue, the last few weeks have been very busy for me.

Issue #809 is relevant and describes that the intended behavior is to not move focus. That allows users to chain the move and focus commands in case they do want to move focus, and just use move in case they don’t.

So, yes, there’s a bug here, but the bug is that focus is moved at all when sending windows to another workspace.

Can you attach a logfile please?

**Comment by mikkeloscar** [comment]
Thanks for taking the time to review this.

I will still argue that the intended behavior is counter intuitive. To me it's very strange that `move left|right|up|down` behaves one way when used on a single screen, but differently when used between multiple screens.

To me `move workspace x` as described in #809 is a completely different command than `move left|right|up|down` and the way it's supposed to work (not moving focus) makes perfect sense.

However, if this is how it is, then I will have to accept it :)

Regarding the log, I suppose any log where I move windows between screens will work, since the default behavior is bugged as I understand from how you describe it, right?

**Comment by stapelberg** [comment]
Yep.

**Comment by acrisci** [comment]
I was the one who implemented `move left|right|up|down` across outputs and my intention was that focus should always follow the window.

**Comment by orestisf1993** [comment]
Closing in favor of #3518: https://github.com/i3/i3/issues/3518#issuecomment-437589026
